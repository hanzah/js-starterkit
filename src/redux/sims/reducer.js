import { handleActions } from 'redux-actions'
import mockSims from './mocks'
import {
  SIMS_SELECTED
} from './constants'

export const initialState = {
  list: mockSims,
	ui: {
		selectedSims: {}
	}
}

export default handleActions({
  [SIMS_SELECTED]: (state, action) => {
    return {
			...state,
			ui: {
				...state.ui,
				selectedSims: action.payload
			}
		}
  }
}, initialState)
