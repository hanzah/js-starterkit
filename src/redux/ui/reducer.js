import { handleActions } from 'redux-actions'

import {
  TOGGLE_MENU
} from './constants'

export const initialState = {
  menuOpened: true
}

export default handleActions({
  [TOGGLE_MENU]: (state) => {
    return {
			...state,
			menuOpened: !state.menuOpened
		}
  }
}, initialState)
