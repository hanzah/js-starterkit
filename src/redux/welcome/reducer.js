import { handleActions } from 'redux-actions'

import {
  CHANGE_MESSAGE
} from './constants'

export const initialState = {
  message: 'Initial message'
}

export default handleActions({
  [CHANGE_MESSAGE]: (state) => {
    return {
			...state,
			message: 'Message Changed'
		}
  }
}, initialState)
