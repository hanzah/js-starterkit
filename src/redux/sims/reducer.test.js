import reducer, { initialState } from './reducer'
import {
  CHANGE_MESSAGE,
	TOGGLE_SIM_STATUS_MODAL
} from './constants'

describe('Welcome Reducer', () => {
	it('should return initial state', () => {
		expect(reducer(initialState, { type: 'test' })).toMatchSnapshot();
	});

	it('should return changed state', () => {
		expect(reducer(initialState, { type: CHANGE_MESSAGE })).toMatchSnapshot();
	});

	it('should return changed sim status modal state after toggle event', () => {
		expect(reducer(initialState, { type: TOGGLE_SIM_STATUS_MODAL })).toMatchSnapshot();
	});
});
