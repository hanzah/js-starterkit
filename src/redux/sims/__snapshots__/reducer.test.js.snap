exports[`Welcome Reducer should return changed state 1`] = `
Object {
  "list": Array [
    Object {
      "country": "USA",
      "ipAddress": "10.70.196.202",
      "network": "AT&T",
      "number": "8901170425230326675",
      "plan": "AT&T Data Rate Plan",
      "sessions": 430,
      "status": "ACTIVE",
    },
    Object {
      "country": "USA",
      "ipAddress": "10.70.196.202",
      "network": "AT&T",
      "number": "8901170425230326676",
      "plan": "AT&T Data Rate Plan",
      "sessions": 430,
      "status": "ACTIVATION READY",
    },
    Object {
      "country": "USA",
      "ipAddress": "10.70.196.202",
      "network": "AT&T",
      "number": "8901170425230326677",
      "plan": "AT&T Data Rate Plan",
      "sessions": 430,
      "status": "INACTIVE",
    },
  ],
  "ui": Object {
    "selectedSims": undefined,
  },
}
`;

exports[`Welcome Reducer should return initial state 1`] = `
Object {
  "list": Array [
    Object {
      "country": "USA",
      "ipAddress": "10.70.196.202",
      "network": "AT&T",
      "number": "8901170425230326675",
      "plan": "AT&T Data Rate Plan",
      "sessions": 430,
      "status": "ACTIVE",
    },
    Object {
      "country": "USA",
      "ipAddress": "10.70.196.202",
      "network": "AT&T",
      "number": "8901170425230326676",
      "plan": "AT&T Data Rate Plan",
      "sessions": 430,
      "status": "ACTIVATION READY",
    },
    Object {
      "country": "USA",
      "ipAddress": "10.70.196.202",
      "network": "AT&T",
      "number": "8901170425230326677",
      "plan": "AT&T Data Rate Plan",
      "sessions": 430,
      "status": "INACTIVE",
    },
  ],
  "ui": Object {
    "selectedSims": Object {},
  },
}
`;
